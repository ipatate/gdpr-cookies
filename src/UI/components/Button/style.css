@import '../../variables.css';

.gdpr_btn {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 0;
    padding: 8px 16px;
    border: 0;
    font-size: 14px;
    color: var(--gdpr-font-black);
    cursor: pointer;
    text-decoration: none;
    text-align: center;

    &-round {
        border-radius: var(--gdpr-radius-bt);
    }

    &-success {
        background: var(--gdpr-color-success);
    }

    &-error {
        background: var(--gdpr-color-danger);
    }

    &-default {
        background: var(--gdpr-color-muted);
        color: var(--gdpr-font-black);
    }

    &:hover {
        opacity: .8;
    }

    &:focus {
        border: 1px dotted var(--gdpr-font-black);
    }
}
