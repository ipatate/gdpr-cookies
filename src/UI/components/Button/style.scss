@import '../../variables.scss';

.gdpr_btn {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 0;
    padding: 8px 16px;
    border: 0;
    font-size: 14px;
    color: $font-white;
    cursor: pointer;
    text-decoration: none;
    text-align: center;

    &-round {
        border-radius: $radius-bt;
    }

    &-success {
        background: $color-success;
    }

    &-error {
        background: $color-danger;
    }

    &-default {
        background: darken($color-muted, 5%);
        color: $font-black;
    }

    &:hover {
        opacity: .8;
    }
}
