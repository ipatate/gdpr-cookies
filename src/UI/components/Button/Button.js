// @flow @jsx h
import {h} from 'preact';
import './style.scss';

type BtProps = {
  onClick: ?Function,
  className: string,
  children: Node,
  'aria-label': ?string,
};
const Button = (props: BtProps) => {
  const {className, onClick, children} = props;
  return (
    <button
      aria-label={props['aria-label'] ? props['aria-label'] : null}
      className={`gdpr_btn ${className}`}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

Button.defaultProps = {
  onClick: () => true,
  className: '',
  'aria-label': null,
};
export default Button;
