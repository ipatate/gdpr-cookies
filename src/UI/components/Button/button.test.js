import {h} from 'preact';
import {render, fireEvent} from '@testing-library/preact';
import Button from './Button';

test('test button with click and class name', () => {
  const onClick = jest.fn();
  const tree = render(
    <Button onClick={onClick} className="test">
      My Button
    </Button>,
  );
  fireEvent(tree.getByText('My Button'), new Event('click'));
  expect(tree.getByText('My Button')).toMatchSnapshot();
  expect(onClick).toHaveBeenCalledTimes(1);
});
