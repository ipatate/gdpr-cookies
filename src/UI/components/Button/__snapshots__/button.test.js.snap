// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<button
  class="gdpr_btn test"
>
  My Button
</button>
`;
